pkg/controller/util.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"errors"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+)
+
+const (
+	pluginName = "fake-plugin-to-gce"
+	attribTrue = "true"
+)
+
+type CSITranslationLibraryInterface interface {
+	TranslateInTreeStorageClassToCSI(inTreePluginName string, sc *storagev1.StorageClass) (*storagev1.StorageClass, error)
+	TranslateCSIPVToInTree(pv *v1.PersistentVolume) (*v1.PersistentVolume, error)
+	IsPVMigratable(pv *v1.PersistentVolume) bool
+	TranslateInTreePVToCSI(pv *v1.PersistentVolume) (*v1.PersistentVolume, error)
+
+	IsMigratedCSIDriverByName(csiPluginName string) bool
+	GetInTreeNameFromCSIName(pluginName string) (string, error)
+}
+
+var _ CSITranslationLibraryInterface = &fakeTranslationLibrary{}
+
+type fakeTranslationLibrary struct{}
```

Comment:
Is this fake specific to `pkg/controller/` in any way?

Reply:
I think this is closer to scenario 1. I don't really know what to name the file though... `fake_translator_test.go` is what i've come up with so far. It's not specific to `pkg/controller` but I forsee it will only be used there in this repo

Reply:
If the fake is only consumed in `controller_test.go` I would not mind its implementation being in that file. If you'd prefer to keep it in a separate file, then `fake_translator_test.go` seems fine.